src/kakarot/execution_context.cairo

```diff
@@ -487,18 +487,22 @@ namespace ExecutionContext {
 
     // @notice Update the array of contracts to destroy.
     // @param self The pointer to the execution context.
-    // @param destroy_contracts_len Array new length.
-    // @param destroy_contracts The pointer to the new array of contracts.
-    func update_destroy_contracts(
+    // @param destroy_contracts_len Array length of destroy_contracts to add.
+    // @param destroy_contracts The pointer to the new array of contracts to destroy.
+    func add_n_destroy_contracts(
```

Comment:
`append_to_destroy_contracts`? `push_to_destroy_contracts` ? non blocking though

Reply:
Let's go for push !